Replace pkg_resources usage

gforcada opened the pull request:
See https://github.com/plone/Products.CMFPlone/issues/4126

gforcada committed:
chore: drop pkg_resources usage

gforcada committed:
Update CHANGES

gforcada committed:
chore: adjust dependencies

davisagli commented on docs/conf.py:
```diff
@@ -51,10 +51,11 @@
-version = '%s.%s' % tuple(rqmt.version.split('.')[:2])
+version = '%s.%s' % tuple(rqmt.split('.')[:2])
 # The full version, including alpha/beta/rc tags.
 release = rqmt.version
```
This will break with `AttributeError: 'str' object has no attribute 'version'` because you already got `.version` on line 21, which returns a `str`

gforcada replied:
Ouch, right, thanks for noticing it!

gforcada commented:
@mauritsvanrees tests are failing due to latest `zc.buildout==4.1` version, if on `tox.ini` one pins `zc.buildout==4.0` then it works

mauritsvanrees reviewed (approved):
LGTM, thanks.

I have just released zc.buildout 4.1.1 with a fix.  If you rerun the failed checks in a few minutes, I think the tests will pass.

mauritsvanrees committed:
Test with latest setuptools and zc.buildout.

In `tox.ini` we were restricting `setuptools < 69` because 70+ gave problems last year.
Those problems have been fixed.  Now we hit a problem in buildout with that old setuptools version:

```
$ tox -e py312
py312: commands_pre[0]> .tox/py312/bin/buildout -nc /Users/maurits/community/plone-coredev/6.2/src/Products.GenericSetup/buildout.cfg buildout:directory=/Users/maurits/community/plone-coredev/6.2/src/Products.GenericSetup/.tox/py312 buildout:develop=/Users/maurits/community/plone-coredev/6.2/src/Products.GenericSetup install test
While:
  Installing.
  Checking for upgrades.

An internal error occurred due to a bug in either zc.buildout or in a
recipe being used:
Traceback (most recent call last):
  File "/Users/maurits/community/plone-coredev/6.2/src/Products.GenericSetup/.tox/py312/lib/python3.12/site-packages/zc/buildout/buildout.py", line 2309, in main
    getattr(buildout, command)(args)
  File "/Users/maurits/community/plone-coredev/6.2/src/Products.GenericSetup/.tox/py312/lib/python3.12/site-packages/zc/buildout/buildout.py", line 717, in install
    self._maybe_upgrade()
  File "/Users/maurits/community/plone-coredev/6.2/src/Products.GenericSetup/.tox/py312/lib/python3.12/site-packages/zc/buildout/buildout.py", line 1165, in _maybe_upgrade
    if not inspect.getfile(sys.modules[project]).startswith(dist.location):
                                                            ^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'location'
```

This is something that may be fixable in `zc.buildout`.
But the easiest fix is to use setuptools 75.8.2 (or a future version).

mauritsvanrees commented:
[This test run](https://github.com/zopefoundation/Products.GenericSetup/actions/runs/13651874935/job/38202364157) passed on Python 3.8, but failed on 3.9 and higher with something like:

```
  File "/Users/maurits/community/plone-coredev/6.2/src/Products.GenericSetup/.tox/py312/lib/python3.12/site-packages/zc/buildout/buildout.py", line 1165, in _maybe_upgrade
    if not inspect.getfile(sys.modules[project]).startswith(dist.location):
                                                            ^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'location'
```

This could be yet another fix in `zc.buildout`.  Or we use `setuptools` 75.8.2, which is better at finding packages.  So that is what I did in commit c4b0ab37838e4924d5060e1f90db415833effbe9.  That leads to [this test run](https://github.com/zopefoundation/Products.GenericSetup/actions/runs/13664487074) which passes for all environments, except Python 3.8.  The reason is simple: neither the lastest zc.buildout nor the latest setuptools works on Python 3.8.

I tried locally without restraints in tox, so just require setuptools, zc.buildout (and wheel) without any minimum version, but then 3.9, 3.10, and 3.11 failed.

So next up: drop Python 3.8 support.  I suppose we are fine with that.
I will run zope.meta here.

mauritsvanrees committed:
Add support for Python 3.13, drop 3.8.

We need this otherwise https://github.com/zopefoundation/Products.GenericSetup/pull/134 won't get green.

This selectively re-applies zope.meta.  I did not want to add a pre-commit config and apply black/isort/etc changes, as this would make the existing PR harder to review.

Bumped version for breaking release.

mauritsvanrees committed:
Temporarily manually put back 'lint' in the gh-actions.

With a full application of zope.meta, this is not needed, because pre-commit.ci does the checks.
But as said in the previous commit, I did not want to do that just now.

mauritsvanrees commented:
The tests are green now.

But the Python 3.8 tests are required and pending.  This should be removed from the branch protection rules now that we have dropped support.  I think I can do this myself, but wanted an approval first.

gforcada commented:
Python 3.8 is being retired in both Zope and Plone, so I would say this is totally fine 👍🏾

dataflake commented:
@gforcada I am confused by the package management related changes. Did you run the zope.meta package configuration script against the repository? If yes then your version of zope.meta must be seriously outdated.

dataflake reviewed (changes requested):
Please remove all packaging-related changes that are handled by zope.meta here, those are bad. Looks like you either ran an outdated version or you just manually changed files that are under zope.meta control.

mauritsvanrees commented:
I replaced this by PR #136, which retains only the relevant commits, and squashes them together.
